src/base/display.jl

```diff
@@ -143,80 +169,117 @@ function Base.show(io::IO, layer::AbstractFeatureLayer)
     nfielddisplay = min(n, 5)
     for i in 1:nfielddisplay
-        fd = getfielddefn(featuredefn, i-1)
-        println(io, "     Field (index $(i-1)): $fd")
+        fd = getfielddefn(featuredefn, i - 1)
+        println(io, "     Field (index $(i - 1)): $fd")
     end
     n > 5 && print(io, "...\n Number of Fields: $n")
+    return io
 end
 
-function Base.show(io::IO, fd::AbstractFieldDefn)
-    fd.ptr == C_NULL && (return print(io, "NULL FieldDefn"))
+function Base.show(io::IO, fd::AbstractFieldDefn)::IO
+    if fd.ptr == C_NULL
+        print(io, "NULL FieldDefn")
+        return io
+    end
     print(io, "$(getname(fd)) ($(gettype(fd)))")
+    return io
 end
 
-function Base.show(io::IO, gfd::AbstractGeomFieldDefn)
-    gfd.ptr == C_NULL && (return print(io, "NULL GeomFieldDefn"))
+function Base.show(io::IO, gfd::AbstractGeomFieldDefn)::IO
+    if gfd.ptr == C_NULL
+        print(io, "NULL GeomFieldDefn")
+        return io
+    end
     print(io, "$(getname(gfd)) ($(gettype(gfd)))")
+    return io
 end
 
-function Base.show(io::IO, feature::Feature)
-    feature.ptr == C_NULL && (return println(io, "NULL Feature"))
+function Base.show(io::IO, feature::Feature)::IO
+    if feature.ptr == C_NULL
+        print(io, "NULL Feature")
+        return io
+    end
     println(io, "Feature")
     n = ngeom(feature)
     for i in 1:min(n, 3)
-        displayname = geomname(getgeom(feature, i-1))
-        println(io, "  (index $(i-1)) geom => $displayname")
+        displayname = geomname(getgeom(feature, i - 1))
+        println(io, "  (index $(i - 1)) geom => $displayname")
     end
     n > 3 && println(io, "...\n Number of geometries: $n")
     n = nfield(feature)
     for i in 1:min(n, 10)
-        displayname = getname(getfielddefn(feature, i-1))
-        print(io, "  (index $(i-1)) $displayname => ")
-        println(io, "$(getfield(feature, i-1))")
+        displayname = getname(getfielddefn(feature, i - 1))
+        print(io, "  (index $(i - 1)) $displayname => ")
+        println(io, "$(getfield(feature, i - 1))")
     end
     n > 10 && print(io, "...\n Number of Fields: $n")
+    return io
 end
 
-function Base.show(io::IO, spref::AbstractSpatialRef)
-    spref.ptr == C_NULL && (return print(io, "NULL Spatial Reference System"))
+function Base.show(io::IO, spref::AbstractSpatialRef)::IO
+    if spref.ptr == C_NULL
+        print(io, "NULL Spatial Reference System")
+        return io
+    end
     projstr = toPROJ4(spref)
     if length(projstr) > 45
         projstart = projstr[1:35]
-        projend = projstr[end-4:end]
+        projend = projstr[(end-4):end]
         print(io, "Spatial Reference System: $projstart ... $projend")
     else
         print(io, "Spatial Reference System: $projstr")
     end
+    return io
 end
 
-function Base.show(io::IO, geom::AbstractGeometry)
-    geom.ptr == C_NULL && (return print(io, "NULL Geometry"))
+function Base.show(io::IO, geom::AbstractGeometry)::IO
+    if geom.ptr == C_NULL
```

Comment:
Is `===` an option to use here? Could be faster according to the [docs](https://docs.julialang.org/en/v1/manual/performance-tips/#Checking-for-equality-with-a-singleton). If yes, this occurs all over the codebase, would require a search/replace.

Reply:
Opened #202 for it.